models/iaf_cond_alpha_mc_fixedca.cpp

```diff
@@ -0,0 +1,853 @@
+ * Iteration function
+ * ---------------------------------------------------------------- */
+
+extern "C" int
+nest::iaf_cond_alpha_mc_fixedca_dynamics( double,
+  const double y[],
+  double f[],
+  void* pnode )
+{
+  // some shorthands
+  typedef nest::iaf_cond_alpha_mc_fixedca N;
+  typedef nest::iaf_cond_alpha_mc_fixedca::State_ S;
+
+  // get access to node so we can work almost as in a member function
+  assert( pnode );
+  const nest::iaf_cond_alpha_mc_fixedca& node =
+    *( reinterpret_cast< nest::iaf_cond_alpha_mc_fixedca* >( pnode ) );
+
+  // compute dynamics for each compartment
+  // computations written quite explicitly for clarity, assume compile
+  // will optimized most stuff away ...
+  for ( size_t n = 0; n < N::NCOMP; ++n )
+  {
+    // membrane potential for current compartment
+    const double V = y[ S::idx( n, S::V_M ) ];
+
+    // excitatory synaptic current
+    const double I_syn_exc =
+      ( y[ S::idx( n, S::G_EXC ) ] ) * ( V - node.P_.E_ex[ n ] );
+    // inhibitory synaptic current
+    const double I_syn_inh =
+      ( y[ S::idx( n, S::G_INH ) ] ) * ( V - node.P_.E_in[ n ] );
+
+    // leak current
+    const double I_L = node.P_.g_L[ n ] * ( V - node.P_.E_L[ n ] );
+
+    double ica = 0.0;
+    if ( n == N::DIST )
+      ica = node.S_.ICa_;
+
+    // coupling currents
+    const double I_conn =
+      ( n > N::SOMA
+          ? node.P_.g_conn[ n - 1 ]
+            * ( ( V - node.P_.E_L[ n ] )
+                - ( y[ S::idx( n - 1, S::V_M ) ] - node.P_.E_L[ n - 1 ] ) )
+          : 0 )
+      + ( n < N::NCOMP - 1
+            ? node.P_.g_conn[ n ]
+              * ( ( V - node.P_.E_L[ n ] )
+                  - ( y[ S::idx( n + 1, S::V_M ) ] - node.P_.E_L[ n + 1 ] ) )
+            : 0 );
+
+    // derivatives
+    // membrane potential
+    f[ S::idx( n, S::V_M ) ] =
+      ( -I_L - I_syn_exc - I_syn_inh - I_conn + node.B_.I_stim_[ n ]
+        + node.P_.I_e[ n ] + ica + y[ S::idx( n, S::I_AP ) ] )
+      / node.P_.C_m[ n ];
+
+    // excitatory conductance
+    f[ S::idx( n, S::DG_EXC ) ] =
+      -y[ S::idx( n, S::DG_EXC ) ] / node.P_.tau_synE[ n ];
+    f[ S::idx( n, S::G_EXC ) ] = y[ S::idx( n, S::DG_EXC ) ]
+      - y[ S::idx( n, S::G_EXC ) ] / node.P_.tau_synE[ n ];
+
+    // inhibitory conductance
+    f[ S::idx( n, S::DG_INH ) ] =
+      -y[ S::idx( n, S::DG_INH ) ] / node.P_.tau_synI[ n ];
+    f[ S::idx( n, S::G_INH ) ] = y[ S::idx( n, S::DG_INH ) ]
+      - y[ S::idx( n, S::G_INH ) ] / node.P_.tau_synI[ n ];
+
+    // active current during AP
+    f[ S::idx( n, S::DI_AP ) ] =
+      -y[ S::idx( n, S::DI_AP ) ] / node.P_.tau_currAP[ n ];
+    f[ S::idx( n, S::I_AP ) ] = y[ S::idx( n, S::DI_AP ) ]
+      - y[ S::idx( n, S::I_AP ) ] / node.P_.tau_currAP[ n ];
+  }
+
+  return GSL_SUCCESS;
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
```

Comment:
Here I would leave a reference to the paper that describes the parameters chosen here, also to refer to a human readable description of their meaning.